### PR TITLE
Feature/loadingscreen refactor

### PR DESCRIPTION
**Pull request by alexanderbock**
Feature to refactor the SceneGraphNode initialization into non-OpenGL and OpenGL parts thus enabling the non-OpenGL parts to be multithreaded
Added loading screen that displays the progress of the loading

### CODE REVIEW

**Comment by alexanderbock** [comment]
Handles issue #411

File: modules/digitaluniverse/rendering/renderablebillboardscloud.cpp [hunk 1]
```diff
@@ -698,14 +698,14 @@ void RenderableBillboardsCloud::render(const RenderData& data, RendererTasks&) {
     glm::vec3 orthoRight = glm::normalize(glm::vec3(worldToModelTransform * glm::vec4(right, 0.0)));
     glm::vec3 orthoUp = glm::normalize(glm::vec3(worldToModelTransform * glm::vec4(up, 0.0)));
 
-
+
```
Comment (emiax) [inline modules/digitaluniverse/rendering/renderablebillboardscloud.cpp]: Some trailing space inserted into this file

File: modules/digitaluniverse/rendering/renderableplanescloud.cpp [hunk 2]
```diff
@@ -347,11 +347,11 @@ RenderablePlanesCloud::RenderablePlanesCloud(const ghoul::Dictionary& dictionary
     _scaleFactor.onChange([&]() {
         _dataIsDirty = true;
     });
-
+  
     if (dictionary.hasKey(LabelFileInfo.identifier)) {
         _labelFile = absPath(dictionary.value<std::string>(
             LabelFileInfo.identifier
-        ));
+            ));
```
Comment (emiax) [inline modules/digitaluniverse/rendering/renderableplanescloud.cpp]: Same here

File: modules/digitaluniverse/rendering/renderablepoints.cpp [hunk 3]
```diff
@@ -362,10 +364,11 @@ namespace openspace {
             GLint positionAttrib = _program->attributeLocation("in_position");
 
             if (_hasColorMapFile) {
+
```
Comment (emiax) [inline modules/digitaluniverse/rendering/renderablepoints.cpp]: Empty line

File: modules/volume/rendering/renderabletimevaryingvolume.h [hunk 4]
```diff
@@ -47,9 +47,9 @@ class RenderableTimeVaryingVolume : public Renderable {
 public:
     RenderableTimeVaryingVolume(const ghoul::Dictionary& dictionary);
     ~RenderableTimeVaryingVolume();
-
-    void initialize() override;
-    void deinitialize() override;
+
```
Comment (emiax) [inline modules/volume/rendering/renderabletimevaryingvolume.h]: trailing

File: src/rendering/loadingscreen.cpp [hunk 5]
```diff
@@ -0,0 +1,706 @@
...
+            _items.end()
+        );
+
+    }
+
+    glEnable(GL_CULL_FACE);
+    glEnable(GL_DEPTH_TEST);
+
+    std::this_thread::sleep_for(RefreshRate);
+    OsEng.windowWrapper().swapBuffer();
```
Comment (emiax) [inline src/rendering/loadingscreen.cpp]: Any particular reason why SGCT's render loop is not used for this?
Is there a risk things may look strange in fisheye rendering for example?

File: src/rendering/loadingscreen.cpp [hunk 6]
```diff
@@ -0,0 +1,706 @@
...
+            _items.begin(),
+            _items.end(),
+            [](const Item& i) {
+                return i.status != ItemStatus::Failed;
+            }
+        ),
+        _items.end()
+    );
+
+    //for (Item& i : _items) {
```
Comment (emiax) [inline src/rendering/loadingscreen.cpp]: Do we need this comment?

File: src/scene/scene.cpp [hunk 7]
```diff
@@ -215,9 +217,64 @@ void Scene::sortTopologically() {
 }
 
 void Scene::initialize() {
```
Comment (emiax) [inline src/scene/scene.cpp]: FYI, In my feature/data-management branch, `Scene::initialize` does not exist in this form. Instead, nodes are initialized whenever they are added to the scene. I'm sure it's possible for these changes to be incorporated in there anyway though.

**Comment by emiax** [comment]
I have looked though the code briefly, and I think it looks neat. As we talked about, some of these changes will be non-trivial to merge with feature/data-management, but I'm sure it will be possible.

File: src/rendering/loadingscreen.cpp (hunk 5 above)
Comment (alexanderbock) [reply src/rendering/loadingscreen.cpp]: Yeah..the initizliation happens before going into the render loop

File: src/rendering/loadingscreen.cpp (hunk 6 above)
Comment (alexanderbock) [reply src/rendering/loadingscreen.cpp]: 💥

File: modules/digitaluniverse/rendering/renderablebillboardscloud.cpp (hunk 1 above)
Comment (alexanderbock) [reply modules/digitaluniverse/rendering/renderablebillboardscloud.cpp]: Was due to a missing merge with `master`.. I had removed the trailing spaces on `master`

File: modules/digitaluniverse/rendering/renderableplanescloud.cpp (hunk 2 above)
Comment (alexanderbock) [reply modules/digitaluniverse/rendering/renderableplanescloud.cpp]: Same

File: modules/digitaluniverse/rendering/renderablepoints.cpp (hunk 3 above)
Comment (alexanderbock) [reply modules/digitaluniverse/rendering/renderablepoints.cpp]: Same

**Pull request merged**